src/node_crypto.cc

```diff
@@ -1025,6 +1066,46 @@ void SecureContext::LoadPKCS12(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+#ifndef OPENSSL_NO_ENGINE
+void SecureContext::SetClientCertEngine(
+  const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  CHECK(args.Length() == 1 && args[0]->IsString());
+
+  SecureContext* sc = Unwrap<SecureContext>(args.This());
+
+  ClearErrorOnReturn clear_error_on_return;
+
+  // If an engine was previously set, free it.
+  // This is because SSL_CTX_set_client_cert_engine does not itself
+  // 'support' multiple calls by cleaning up before overwriting the
+  // client_cert_engine internal context variable
+  if (sc->ctx_->client_cert_engine != NULL) {
+    ENGINE* e = sc->ctx_->client_cert_engine;
+    sc->ctx_->client_cert_engine = nullptr;
+    ENGINE_finish(e);
+  }
+
+  // Load engine
+  const node::Utf8Value engine_id(env->isolate(), args[0]);
+  ENGINE* engine = LoadEngineById(env, *engine_id);
```

Comment:
It's not a deal breaker but `LoadEngineById()` currently conflates two things, that's why I suggested removing the dependency on `env`.

> The thing that hurts my eye with that solution is the arbitrary constant 1024. And that you have to allocate 1024 bytes to call a function in case it fails...

It's 1024 bytes on the stack so it's really just a frame pointer bump (and, if I'm being pedantic, almost certainly subsumed by the existing stack frame.)

As to 1024 being arbitrary constant, that's true, but I'm not disconcerted by it.  Now, if it was 1025, that _would_ disconcert my programmer sensibilities...

> Perhaps it would be more consistent to use exceptions in the C++ layer and map it 1-1 with the JS exceptions?

Node is compiled with `-fno-exceptions` so that's not an option.